src/core/components/ModelDropdownExpanded.tsx

```diff
@@ -0,0 +1,66 @@
+import styles from './NavBar.module.css';
+import { getAllModels, isBlockchainModelName } from '../config/models';
+import { ModelConfig } from '../types/models';
+import { useAppContext } from '../context/useAppContext';
+
+interface ModelDropdownProps {
+  setDropdownOpen: (dropdownOpen: boolean) => void;
+}
+
+import { useEffect, useState } from 'react';
+
+export const useMediaQuery = (query: string) => {
+  const [matches, setMatches] = useState<boolean>(
+    window.matchMedia(query).matches,
+  );
+
+  useEffect(() => {
+    const matchQuery = window.matchMedia(query);
+    const onChange = (e: MediaQueryListEvent) => {
+      setMatches(e.matches);
+    };
+
+    matchQuery.addEventListener('change', onChange);
+    return () => matchQuery.removeEventListener('change', onChange);
+  }, [query]);
+
+  return matches;
+};
+
+export const useIsMobile = () => useMediaQuery('(max-width: 768px)');
+
+const ModelDropdownExpanded = ({ setDropdownOpen }: ModelDropdownProps) => {
+  const { handleModelChange } = useAppContext();
+
+  const isMobile = useIsMobile();
+
+  const handleModelClick = (model: ModelConfig) => {
+    handleModelChange(model.id);
+    setDropdownOpen(false);
+  };
+  return (
+    <div className={styles.dropdownMenu} data-testid="model-dropdown-menu">
+      {getAllModels().map((model) => {
+        const ModelOptionIcon = model.icon;
+
+        //  since we don't support metamask on mobile, don't allow a user to switch to blockchain models
+        const disableOption = isMobile && isBlockchainModelName(model.id);
```

Comment:
Keep users from choosing a blockchain model if they are on mobile until we support a mobile wallet.